### PR TITLE
Fix environment variable propagation

### PR DESCRIPTION
**Pull request by Tabrizian**
Fixes the environment variable propagation issue mentioned in triton-inference-server/server#2157.

Should be merged after #16.

### CODE REVIEW

**Comment by deadeyegoodwin** [comment]
@Tabrizian  Please rebase so that is has only the env changes.

**Comment by Tabrizian** [comment]
Rebased.

File: src/python.cc [hunk 1]
```diff
@@ -234,7 +238,7 @@ TRITONSERVER_Error *ModelInstanceState::CreatePythonInterpreter() {
     subinterpreter_commandline[5] = pymodule_path_.c_str();
     subinterpreter_commandline[7] = name_.c_str();
     if (execvpe(subinterpreter_commandline[0],
```
Comment (deadeyegoodwin) [inline src/python.cc:240]: Since you are just passing in the existing env you can just use execvp and not need to use "environ"

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: src/python.cc (hunk 1 above)
Comment (Tabrizian) [reply src/python.cc]: fixed. Also, @tanmayv25 pointed out the reason the formatting was inconsistent. I fixed that and reformatted the whole document.

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**